react/features/overlay/components/AbstractOverlay.js

```diff
@@ -0,0 +1,71 @@
+/* global $, APP */
+import React, { Component } from 'react';
+
+/**
+ * Implements an abstract React Component for overlay - the components which
+ * are displayed on top of the application covering the whole screen.
+ */
+export default class AbstractOverlay extends Component {
+    /**
+     * Initializes a new AbstractOverlay instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     * @public
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            /**
+             * Indicates the css style of the overlay. if true - lighter  and
+             * darker otherwise.
+             * @type {boolean}
+             */
+            isLightOverlay: false
```

Comment:
I think it's good to keep it in the state because this is abstract class and in future some of the classes that are going to extend it may need to change during their lifetime.